_posts/2020/2020-02-11-unit-testing-in-react.md

```diff
@@ -0,0 +1,251 @@
+---
+layout: post
+title: Unit testing in React
+date: 2020-02-11
+header_image: public/code-coding-computer-cyberspace-270373.jpg
+header_overlay: true
+category: coding
+tags: ["javascript", "reactjs", "testing"]
+authors: ["Daniel"]
+about_authors: ["dhara"]
+---
+
+Automated Testing is one of the most important aspects of software development. You can test your code in different ways and levels, the most relevant
+of which are unit testing and integration testing.
+On the one hand, in unit testing, you test each one of your functions and components individually. On the other, in integration testing, you test
+how all those work along with each other to deliver all the fancy features you offer your users.
+
+As Adrià Fontcuberta pointed out at his remarkable talk at dotJS in Paris (check out this [blog post](/blog/events/dotjs-2019-in-paris-from-the-perspective-of-a-frontend-designer/){:target="_blank"} about the conference, as well as the [video](https://www.dotconferences.com/2019/12/adria-fontcuberta-the-pragmatic-front-end-tester){:target="_blank"} of the actual talk), tests give predictability and
+"help us sleep well at night".
+
+In this blog post, I'll show you useful hints for unit testing a simple component in React.
+
+## Getting started
+
+One of our most recent features at our storefront was a "Load more" button in the product pages. We could draft here a simplified version of this
+feature, and carry on to writing tests for it! This will help us to understand how tests work and how to ensure our tests are making our code
```

Comment:
Sugesstion: "... and how to ensure that ..."

Reply:
In this case, I guess it is indifferent. I find the variant without "that" more direct and closer to spoken informal dialogue.